packages/toolpad-app/src/toolpad/AppEditor/PageEditor/JsExpressionEditor.tsx

```diff
@@ -41,25 +44,45 @@ export function JsExpressionEditor({
     const type = jsonToTs(globalScope);
 
     const globals = Object.keys(globalScope)
-      .map((key) => `declare const ${key}: RootObject[${JSON.stringify(key)}];`)
-      .join('\n');
+      .map((key) => {
+        let declaration = ``;
+        const metaData = globalScopeMeta[key] || {};
 
-    // TODO: cleanup after Toolpad v1
-    const queryDeprecation = `
-      /** @deprecated use parameters */
-      declare const query: Query;
-    `;
+        if (metaData.description) {
+          declaration = `/** ${metaData.description} */`;
```

Comment:
I believe they all need to go in the same comment block to support the case of having both a description and a comment defined

Reply:
`deprecated` is defined as `boolean | string` so if you want to add description of the deprecation then you should use it like this:
```
{
  query: {
    deprecated: 'Reason of deprecation'
  }
}
```
or do you mean using deprecation message + description?
Would we even use both at the same time?

Reply:
> or do you mean using deprecation message + description?

yes

> Would we even use both at the same time?

The `description` would describe the global variable. The `deprecation` would describe why it's deprecated or what to use instead.